File: arrays/encryption.py
import math

str = "feed the dog"
str = str.replace(' ','')
strlength = len(str)
sqrt = math.sqrt(strlength)
row = math.floor(sqrt)
column = math.ceil(sqrt)

# ROW x COL check
if row*column<strlength:
  row = math.ceil(sqrt)

def encryption(str):
  arr = []
  for i in range(row):
    colList = []
    for j in range(column):
      if (column*i)+j<len(str):
        colList.append(str[(column*i)+j])
    arr.append(colList)
  
  print(arr)
  newStr =''
  for i in range(column):
    for j in range(row):
      if i<len(arr[j]):
        newStr += arr[j][i]
    newStr+=' '

  print(newStr)

encryption(str)

File: arrays/encryption.js
// O(r*c) => r-row of matris, c-column of matris

function encryptString(str){
    str = str.replace(/\s+/g, '');
    strLen = str.length;
    sqrt = Math.sqrt(strLen);
    row = Math.floor(sqrt);
    column = Math.ceil(sqrt);
  
    if((row * column) < strLen){
      row = Math.ceil(sqrt)
    }
  
    let arr = [];
  
    for(let i=0;i<row;i++){
      let stringof = [];
      for(let j=0;j<column;j++){
        if((column*i)+j< strLen){
          stringof.push(str.charAt((column*i) + j))
        }
      }
      arr.push(stringof);
    }
  
    let newstr = "";
    for(let c=0;c<column;c++){
      //newstr = newstr + " ";
      for(let r=0;r<row;r++){
        if(c<arr[r].length){
          newstr = newstr + arr[r][c];
        } 
  
      }
      newstr = newstr + " ";
    }
  
    console.log(arr);
    console.log(newstr);
  
  }
  encryptString("this is a rest")